examples/all-clusters-app/all-clusters-common/src/mode-select-delegates.cpp

```diff
@@ -0,0 +1,171 @@
+
+CHIP_ERROR ModeSelectDelegate::Init()
+{
+    return CHIP_NO_ERROR;
+}
+
+Status ModeSelectDelegate::HandleChangeToMode(uint8_t mode)
+{
+    return Status::Success;
+}
+
+void ModeSelectDelegate::HandleChangeToModeWitheStatus(uint8_t mode, Commands::ChangeToModeResponse::Type &response)
+{
+    uint8_t currentMode;
+    EmberAfStatus status = Attributes::CurrentMode::Get(0x1, &currentMode);
+
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        response.status = int8_t(ChangeToModeResponseStatus::kGenericFailure);
+        response.statusText.SetValue(chip::CharSpan("Could not get the current mode", 30));
+        return;
+    }
+
+    // We add our business logic that doesn't allow a transition from mode 0 to 7. If this is attempted, a descriptive
+    // response is sent.
+    if (currentMode == 0 && mode == 7)
+    {
+        response.status = int8_t(ChangeToModeResponseStatus::kGenericFailure);
+        response.statusText.SetValue(chip::CharSpan("Cannot change to mode 7 from mode 0", 35));
+        return;
+    }
+}
+
+//-- RVC Run delegate functions
+CHIP_ERROR RvcRunDelegate::Init()
+{
+    return CHIP_NO_ERROR;
+}
+
+Status RvcRunDelegate::HandleChangeToMode(uint8_t mode)
+{
+    return Status::UnsupportedCommand;
+}
+
+void RvcRunDelegate::HandleChangeToModeWitheStatus(uint8_t mode, Commands::ChangeToModeResponse::Type &response)
+{
+    uint8_t currentMode;
+    EmberAfStatus status = RvcRun::Attributes::CurrentMode::Get(0x1, &currentMode);
+
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        response.status = int8_t(ChangeToModeResponseStatus::kGenericFailure);
+        response.statusText.SetValue(chip::CharSpan("Could not get the current mode", 30));
+        return;
+    }
+
+    // Our business logic states that we can only switch into the mapping state from the idle state.
+    if (mode == 2 && currentMode != 0)
+    {
+        response.status = int8_t(ChangeToModeResponseStatus::kGenericFailure);
+        response.statusText.SetValue(chip::CharSpan("Change to the mapping state is only allowed from idle", 53));
+    }
+
+    response.status = int8_t(ChangeToModeResponseStatus::kSuccess);
+}
+
+//-- RVC Clean delegate functions
+CHIP_ERROR RvcCleanDelegate::Init()
+{
+    return CHIP_NO_ERROR;
+}
+
+Status RvcCleanDelegate::HandleChangeToMode(uint8_t mode)
+{
+    return Status::UnsupportedCommand;
+}
+
+void RvcCleanDelegate::HandleChangeToModeWitheStatus(uint8_t mode, Commands::ChangeToModeResponse::Type &response)
+{
+    uint8_t currentMode;
+    EmberAfStatus status = RvcRun::Attributes::CurrentMode::Get(0x1, &currentMode);
+
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        response.status = int8_t(ChangeToModeResponseStatus::kGenericFailure);
+        response.statusText.SetValue(chip::CharSpan("Could not get the current mode", 30));
+        return;
+    }
+
+    if (currentMode == 1)
+    {
+        response.status = int8_t(RvcClean::ChangeToModeResponseStatus::kCleaningInProgress);
```

Comment:
Please use `static_cast` everywhere to follow common style, rather than `int8_t()` constructor format.

Reply:
Updated in commit [d616e4b7823c2dab11845c55dfcbf2ae4ae3c773](https://github.com/project-chip/connectedhomeip/pull/26447/commits/d616e4b7823c2dab11845c55dfcbf2ae4ae3c773).